code/modules/fish/fish_tank.dm

```diff
@@ -0,0 +1,711 @@
+			examine_message += "<span class='notice'>There's some food in [src], but no fish!</span>"
+		else
+			examine_message += "<span class='warning'>There is no food in \the [src].</span>"
+	else										//We've got fish, report the food level
+		switch (food_level/MAX_FOOD)
+			if(0)
+				examine_message += "<span class='warning'>The fish look very hungry!</span> "
+			if(0.001 to 0.2)
+				examine_message += "<span class='warning'>The fish are nibbling on the last of their food.</span>"
+			if(0.2 to 0.999)				//Breeding is possible
+				examine_message += "<span class='notice'>The fish seem happy!</span>"
+			if(1)
+				examine_message += "<span class='notice'>There is a solid layer of fish food at the top.</span>"
+
+	//Report the number of harvestable eggs
+	if(egg_list.len)								//Don't bother if there isn't any eggs
+		examine_message += "<span class='notice'><br>There are [egg_list.len] eggs able to be harvested!</span>"
+
+	examine_message += "<br>"
+
+	//Report the number and types of live fish if there is water in the tank
+	if(!fish_list.len)
+		examine_message += "<span class = 'warning'>\The [src] doesn't contain any live fish. </span>"
+	else
+		//Build a message reporting the types of fish
+		var/message = "You spot "
+		for (var/i = 1 to fish_list.len)
+			if(fish_list.len > 1 && i == 1)	//If there were at least 2 fish, and this is the last one, add "and" to the message
+				message += "and "
+			message += "a [fish_list[i]]"
+			if(i < fish_list.len)					//There's more fish, add a comma to the message
+				message +=", "
+		message +="."								//No more fish, end the message with a period
+		//Display the number of fish and previously constructed message
+		examine_message += "<span class = 'notice'>\The [src] contains [fish_list.len] live fish. [message]</span>"
+
+
+	//Report lid state for tanks and wall-tanks
+	if(has_lid)									//Only report if the tank actually has a lid
+		examine_message += "<br>"
+		//Report lid state
+		if(lid_switch)
+			examine_message += "<span class = 'notice'>The lid is closed.</span>"
+		else
+			examine_message += "<span class = 'notice'>The lid is open.</span>"
+
+	examine_message += "<br>"
+
+	//Report if the tank is leaking/cracked
+	if(water_level > 0)
+		switch (leaking)					//Tank has water, so it's actually leaking
+			if(MINOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is leaking.</span>"
+			if(MAJOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is leaking profusely!</span>"
+	else
+		switch (leaking)									//No water, report the cracks instead
+			if(MINOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is cracked.</span>"
+			if(MAJOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is nearly shattered!</span>"
+
+
+	//Finally, report the full examine_message constructed from the above reports
+	to_chat(user, "[jointext(examine_message, "")]")
+	return examine_message
```

Comment:
why bother?

Reply:
Could be useful if someone wanted to integrate this into a fishtank monitor or something later.

Reply:
Surely it would be better to split the logic into a different proc in that case